app/naive_bayes.py

```diff
@@ -0,0 +1,286 @@
+
+class LogicalPlanner:
+    """
+    The Naive Bayes Planner utilizes past operational history to execute operations while prioritizing likelihood of link success.
+    With sufficient local data, NB planner runs all operations in order of most to least effective links, while dropping links with insufficient likelihood of success.
+
+    The NB Planner recieves the following parameters:
+        - min_link_data: the minimum number of past runs of a link in existing past operational 
+        data necessary for planner to utilize its probability of success. Can be custom set by user in 
+        \stockpile\data\planners\48e1a882-1606-4910-8f2d-2352eb80cba2.yml, default is 3.
+
+        -  min_prob_link_success: minimum calculated likelihood of success necessary to run a link. Set by user through
+        visibility setting in advanced section of operation creation. Calculated as = (99.0% - operation visibility%). Default is 48%
+
+    Algorithm of the planner:
+        -While there are available links in the operation for live agents:
+            -If there are links that satisfy minimum operational data settings and minimum link probability of success settings:
+                -Execute them in order of likelihood of success (high to low)
+            -Run all remaining links with insufficient past operational data in atomic ordering (used by atomic planner)
+            -Drop any links that with sufficient past data but insufficient calculated probability of success
+        -Terminate when out of links or all links have too low likelihood of success.
+    """
+
+    def __init__(
+        self, 
+        operation: Operation, 
+        planning_svc: PlanningService, 
+        min_link_data : int, 
+        decision_logging : bool,
+        stopping_conditions: List[Fact] = ()
+    ):
+        """
+        :param operation:
+        :param planning_svc:
+        :param min_link_data: Minimum number runs of link required to use past statistics on it
+        :param stopping_conditions:       
+        """
+        self.operation = operation
+        self.planning_svc = planning_svc
+        self.stopping_conditions = stopping_conditions
+        self.stopping_condition_met = False
+        self.state_machine = ['bayes_state']
+        self.next_bucket = 'bayes_state'
+        self.matrix_past_links = None
+        self.min_link_data = min_link_data
+        self.decision_logging = decision_logging
+        self.min_prob_link_success = (99.0-self.operation.visibility)/(100)
+        self.links_executed = 0
+        if self.decision_logging:
+            self.planning_svc.log.debug("Operation started")   
+
+    async def execute(self):
+        await self.planning_svc.execute_planner(self)
+    
+    async def bayes_state(self):
+        """
+        Update past operational data matrix if needed then run top priority link for each agent until operation concluded
+        """
+        if self.links_executed % 10 == 0:
```

Comment:
Just curious, what is the reason for it only recomputing the past link data every 10 links executed? Would this possibly be something we might want to parameterize?